### ISSUE TITLE
[FEATURE] Change the name of the generated model class withe name of message name.

### ISSUE THREAD

**Issue by AayushSaini101** [asyncapi/modelina#1996]
### Why do we need this improvement?

**Asyncapi.yml:**
```
asyncapi: '2.2.0'
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    subscribe:
      message:
        $ref: '#/components/messages/UserSignedUp'
components:
  messages:
    UserSignedUp:
      payload:
        type: object
        properties:
          displayName:
            type: string
            description: Name of the user
          email:
            type: string
            format: email
            description: Email of the user
```
Output:
```
asyncapi generate models java ./asyncapi.yml --packageName=src

Successfully generated the following models: 
## Model name: AnonymousSchema_1 // Name should be change to userSignedUp
package src;

import java.util.Map;
public class AnonymousSchema_1 {
  private String displayName;
  private String email;
  private Map<String, Object> additionalProperties;

  public String getDisplayName() { return this.displayName; }
  public void setDisplayName(String displayName) { this.displayName = displayName; }

  public String getEmail() { return this.email; }
  public void setEmail(String email) { this.email = email; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}
```

### How will this change help?

This change will help to adhere proper naming of the project, lets suppose if we want to call the model class we need to call the AnonymousSchema_1 which seems to be not valid.

### How could it be implemented/designed?

* Use the name of the message as the class name

### 🚧 Breaking changes

No

### 👀 Have you checked for similar open issues?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md)

### Are you willing to work on this issue?

Yes I am willing to submit a PR!

**Comment by jonaslagoni** [comment]
Makes sense, taget `next` @AayushSaini101 :v:

**Comment by black-snow** [comment]
I just stumbled over this. I ran `asyncapi generate models` and end up with:

> Successfully generated the following models: AnonymousSchema1, AnonymousSchema25, AnonymousSchema28, AnonymousSchema31, AnonymousSchema38, AnonymousSchema41, AnonymousSchema44

I would've expected that the file name as well as the class name do use the actual name of the component / message / whatevs. Is there currently no way or do I have to set some extra annotation somewhere?

**Comment by jonaslagoni** [comment]
Happy to accept and help guide a PR 🙂

**Comment by black-snow** [comment]
What annotation would that be?

**Comment by black-snow** [comment]
This would be a breaking change, right?

Any idea where to start? I imagined this to be an easy change but grepping for `anonymous` yields only CommonModel and AsyncAPIInputProcessor apart from the tests / examples. It's not apparent to me where AnonymousSchema_N gets generated.

**Comment by jonaslagoni** [comment]
> What annotation would that be?

You would have to explicitly define the message payload as the name of the message.

> This would be a breaking change, right?

Yes, just target the `next` branch 🙂 Unless you want to hide it behind an option.

> Any idea where to start? I imagined this to be an easy change but grepping for anonymous yields only CommonModel and AsyncAPIInputProcessor apart from the tests / examples. It's not apparent to me where AnonymousSchema_N gets generated.

All this logic happens in the input processor for AsyncAPI: https://github.com/asyncapi/modelina/blob/9f9fc78e40bf1efcc64d73d03085de75d6012976/src/processors/AsyncAPIInputProcessor.ts#L128

**Comment by black-snow** [comment]
Still looking into this - just gotta do it sometime.

**Comment by black-snow** [comment]
Stale yes, but still as relevant as on day one.

**Comment by hartmanjan1970** [comment]
I run also in this bug, is there some progress on this issue?

**Comment by jonaslagoni** [comment]
Does not have the bandwidth to fix it unfortunately... Still happy to accept PRs or help clear stuff up :v:

**Comment by ThomasVerhoeven1998** [comment]
If you define it like this with async api 3.0.0, the generated code is correct
```
asyncapi: 3.0.0
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    address: user/signedup
    messages:
      subscribe.message:
        $ref: '#/components/messages/UserSignedUp'
operations:
  user/signedup.subscribe:
    action: send
    channel:
      $ref: '#/channels/user~1signedup'
    messages:
      - $ref: '#/channels/user~1signedup/messages/subscribe.message'
components:
  schemas:
    UserSignedUp:
      type: object
      properties:
        displayName:
          type: string
          description: Name of the user
        email:
          type: string
          format: email
          description: Email of the user
  messages:
    UserSignedUp:
      payload:
        schemaFormat: application/vnd.aai.asyncapi+json;version=3.0.0
        schema:
          $ref: '#/components/schemas/UserSignedUp'

```

You will get this: 

```
import java.util.Objects;
import java.util.List;
import com.fasterxml.jackson.annotation.*;
import javax.validation.constraints.*;
import javax.validation.Valid;
public class UserSignedUp {
  @JsonProperty("displayName")
  @JsonInclude(JsonInclude.Include.NON_NULL)
  private java.lang.String displayName;
  @JsonProperty("email")
  @JsonInclude(JsonInclude.Include.NON_NULL)
  private java.lang.String email;

  /**
   * Name of the user
   */
  public java.lang.String getDisplayName() { return this.displayName; }
  public void setDisplayName(java.lang.String displayName) { this.displayName = displayName; }

  /**
   * Email of the user
   */
  public java.lang.String getEmail() { return this.email; }
  public void setEmail(java.lang.String email) { this.email = email; }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    UserSignedUp self = (UserSignedUp) o;
      return 
        Objects.equals(this.displayName, self.displayName) &&
        Objects.equals(this.email, self.email);
  }

  @Override
  public int hashCode() {
    return Objects.hash((Object)displayName, (Object)email);
  }
```

**Comment by black-snow** [comment]
@ThomasVerhoeven1998 is that a Java-only thing or what did you change? I cannot spot any extra annotations. For me it's still AnonymousSchemaXY.

**Comment by jonaslagoni** [comment]
> [@ThomasVerhoeven1998](https://github.com/ThomasVerhoeven1998) is that a Java-only thing or what did you change? I cannot spot any extra annotations. For me it's still AnonymousSchemaXY.

@black-snow it's basically just to use the components sections to define each schema and message through that, i.e. `#/components/schemas/UserSignedUp` and `#/components/messages/UserSignedUp` that should set the right names

**Comment by black-snow** [comment]
That's weird - didn't do the trick for me. But then again I'm still on an older version due to the alias thingy. Will report back once I've upgraded.

**Comment by PainOchoco** [comment]
I had the same issue and moving the payload schema definition of a message defined in #/components/messages to #/components/schemas worked for me.

**Comment by hartmanjan1970** [comment]
We have the example of @ThomasVerhoeven1998 working, the only issue is that we can generate only 1 message per channel.
Seems to me that a channel is not bounded by one single message.

Now is my question:
Are there some solutions for generating multiple message per channel?

**Comment by jonaslagoni** [comment]
@hartmanjan1970 right now Modelina just generate it for each operation and their messages. Indirectly what is defined in channels: https://github.com/asyncapi/modelina/blob/c7a6530d125b63820118efb7437eebf9dff98dc8/src/processors/AsyncAPIInputProcessor.ts#L169

TLDR:
```
asyncapi: 3.0.0
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    address: user/signedup
    messages:
      subscribe.message:
        $ref: '#/components/messages/UserSignedUp'
      publish.message:
        $ref: '#/components/messages/UserSignedUp'
operations:
  user/signedup.subscribe:
    action: send
    channel:
      $ref: '#/channels/user~1signedup'
    messages:
      - $ref: '#/channels/user~1signedup/messages/subscribe.message'
      - $ref: '#/channels/user~1signedup/messages/publish.message'
components:
  schemas:
    UserSignedUp:
      type: object
      properties:
        displayName:
          type: string
          description: Name of the user
        email:
          type: string
          format: email
          description: Email of the user
  messages:
    UserSignedUp:
      payload:
        schemaFormat: application/vnd.aai.asyncapi+json;version=3.0.0
        schema:
          $ref: '#/components/schemas/UserSignedUp'
```

So yea, it only generates messages that are intended to be used.

**Comment by black-snow** [comment]
It still only generates `AnonymousSchemaXY` for me (pydantic) - what am I missing? o_Ô

**Comment by fredagsfys** [comment]
FWIW, I played around today and found out that adding messageTraits with schema references BREAKS the naming. The AsyncAPI CLI bug is triggered by messageTraits that reference schemas.

**Comment by ThomasVerhoeven1998** [comment]
> We have the example of [@ThomasVerhoeven1998](https://github.com/ThomasVerhoeven1998) working, the only issue is that we can generate only 1 message per channel. Seems to me that a channel is not bounded by one single message.
> 
> Now is my question: Are there some solutions for generating multiple message per channel?

Shouldn't that be fixed by https://github.com/asyncapi/modelina/pull/2352 @jonaslagoni ?

**Comment by jonaslagoni** [comment]
@ThomasVerhoeven1998 good point... Ill have a look later

**Comment by black-snow** [comment]
Now for pydantic and asyncapi `5.0.6` I get some named models and a bunch of anons, still. Looking at them I can't quite spot the pattern. My payloads get turned into `<propname_object.py>` and class but all others don't. Even though they both live under `.components.schemas`.

```yaml
components:
  messages:
    A:
      title: "A"
      name: "A"
      contentType: "application/json"
      payload:
        allOf:
          # turned into anon
          - $ref: "#/components/schemas/Envelope"
          # turned into anon
          - $ref: "#/components/schemas/EnvelopeOverrides"
          # turned into P_object
          - type: object
            properties:
              data:
                $ref: "#/components/schemas/P"
```
